app/interactors/tasks/destroy/send_notification.rb

```diff
@@ -0,0 +1,21 @@
+module Tasks
+  class Destroy
+    class SendNotification
+      include Interactor
+
+      delegate :project, :task, :user, to: :context
+
+      def call
+        mail_owner.deliver_later unless ProjectMembership.find_by(project: project, user: user).owner?
```

Comment:
Maybe if owner, not unless?

Reply:
it handles logic of sending message to owner that member of his project deleted task so unless should be used to check if user isnt the owner himself, i could've done this check inside mailer but i chose not to

Reply:
Then you can move checking user role to method "initiator_is_owner?"